README: Add Oasis ROFL TEE instructions (#1)
matevz/game-node: game-starter/README.md

    `docker compose up -d`
**Note** We recommend using nvm version 23 `nvm use 23`

## To run project in TEE

### Phala

1. Build the docker image and publish it to the docker hub
    `docker compose build -t <your-dockerhub-username>/virtuals-game-starter`
    `docker push <your-dockerhub-username>/virtuals-game-starter`
2. Deploy to Phala cloud using [tee-cloud-cli](https://github.com/Phala-Network/tee-cloud-cli) or manually with the [Cloud dashboard](https://cloud.phala.network/).
3. Check your agent's TEE proof and verify it on the [TEE Attestation Explorer](https://proof.t16z.com/).

### Oasis ROFL

1. To ROFLize your agent, fetch the [Oasis CLI](https://github.com/oasisprotocol/cli/releases)
   for your OS of choice and run `oasis rofl init`

> 2. Adapt `docker-compose.yaml` as follows:

Reviewer: Currently it also must be named `compose.yaml`. We can change that and use autodetection during `init` (it is configured under the `artifacts` section in the manifest).

   - replace `/var/run/tappd.sock:/var/run/tappd.sock` with

>      `/run/rofl-appd.sock:/run/rofl-appd.sock`

Reviewer: This seems suboptimal.

    `docker compose up -d`
**Note** We recommend using nvm version 23 `nvm use 23`

## To run project in TEE

### Phala

1. Build the docker image and publish it to the docker hub
    `docker compose build -t <your-dockerhub-username>/virtuals-game-starter .`
    `docker push <your-dockerhub-username>/virtuals-game-starter`
2. Deploy to Phala cloud using [tee-cloud-cli](https://github.com/Phala-Network/tee-cloud-cli) or manually with the [Cloud dashboard](https://cloud.phala.network/).
3. Check your agent's TEE proof and verify it on the [TEE Attestation Explorer](https://proof.t16z.com/).

### Oasis ROFL

> 1. To ROFLize your agent, fetch the [Oasis CLI](https://github.com/oasisprotocol/cli/releases)

Reviewer 2: ```suggestion
1. To ROFLize your agent, get the [Oasis CLI](https://github.com/oasisprotocol/cli/releases)
```

>    for your OS of choice and run `oasis rofl init`

Reviewer 2: ```suggestion
   for your OS of choice and run: `oasis rofl init`
```

2. Adapt `docker-compose.yaml` as follows:
   - replace `/var/run/tappd.sock:/var/run/tappd.sock` with
     `/run/rofl-appd.sock:/run/rofl-appd.sock`
   - replace `image:` value to include a fully qualified domain name of your
     favorite container registry (e.g. docker.io, ghcr.io), for
     example:
     `docker.io/<your-dockerhub-username>/virtuals-game-starter`
3. Build the docker image and publish it, for example:

>    `docker build -t docker.io/<your-dockerhub-username>/virtuals-game-starter .`

Reviewer 2: Use enumeration for the 2 example commands to make it clearer.

   `docker push docker.io/<your-dockerhub-username>/virtuals-game-starter`

> 4. Register a new ROFL app on-chain.

Reviewer 2: ```suggestion
4. Register a new ROFL app on-chain: `oasis rofl create`
```

And similarly below. Add `:` before referencing CLI commands.

   `oasis rofl create`
5. Build the ROFL bundle and let it update the manifest file with the enclave ID
   `oasis rofl build --update-manifest`
6. Deploy your ROFL to your provider on the network:
   `oasis rofl deploy`

> 7. Check that your instance is up and running:

Reviewer 2: ```suggestion
7. Check that your ROFL instance is up and running:
```

   `oasis rofl show`

> 8. You can verify whether the currently deployed instance really corresponds to

Reviewer 2: ```suggestion
8. (OPTIONAL) Anyone can independently verify whether the currently deployed ROFL instance really corresponds to
```

   
    `docker compose build -t <your-dockerhub-username>/virtuals-game-starter .`

    `docker push <your-dockerhub-username>/virtuals-game-starter`

2. Deploy to Phala cloud using [tee-cloud-cli](https://github.com/Phala-Network/tee-cloud-cli) or manually with the [Cloud dashboard](https://cloud.phala.network/).

3. Check your agent's TEE proof and verify it on the [TEE Attestation Explorer](https://proof.t16z.com/).

### Oasis ROFL

1. To ROFLize your agent, get the [Oasis CLI](https://github.com/oasisprotocol/cli/releases)
   for your OS of choice.

2. Register a new ROFL app and encrypt the secrets inside the `rofl.yaml`
   manifest file:
   
     ```shell
     oasis rofl create
     echo -n "your_game_api_key_here" | oasis rofl secret set API_KEY -
     echo -n "your_openai_api_key_here" | oasis rofl secret set OPENAI_API_KEY -
     echo -n "your_weather_api_key_here" | oasis rofl secret set WEATHER_API_KEY -
     ```

3. Build the docker image and publish it, for example:
   
   ```shell
   docker build -t docker.io/<your-dockerhub-username>/virtuals-game-starter .
   docker push docker.io/<your-dockerhub-username>/virtuals-game-starter
   ```
   

>    In `rofl-compose.yml` replace `<your-dockerhub-username>` with your acutal

Reviewer: ```suggestion
   In `rofl-compose.yml` replace `<your-dockerhub-username>` with your actual
```

>    one. For security, we strongly suggest to hardcode the image hash reported

Reviewer: ```suggestion
   one. To ensure integrity, we strongly suggest to hardcode the image hash reported
```

   when you pushed the image. For example:
   
   ```yaml
   services:
     game-starter:
       image: docker.io/rosy/virtuals-game-starter@sha256:25263747e8f9ebc193e403ac009b696ea49459d9d642b86d890de432dae4469f
   ```
   
4. Build the bundle, submit the obtained Enclave ID and the secrets to
   the chain and deploy it:
   
   ```shell

>    oasis rofl build --update-manifest

Reviewer: Probably should release the new CLI first and then omit the `--update-manifest`?
